### PR TITLE
feat(insights-caching): Integrate new caching system

### PR DESCRIPTION
**Pull request by macobo**
Follow-up to https://github.com/PostHog/posthog/pull/13127

This PR removes the old celery tasks and cache refreshing code and replaces it with the new.

### CODE REVIEW

File: posthog/api/insight.py [hunk 1]
```diff
@@ -375,20 +375,16 @@ def get_last_refresh(self, insight: Insight):
         result = self.get_result(insight)
 
         if result is not None:
+            caching_state = dashboard_tile.caching_state if dashboard_tile else insight.caching_state
```
Comment (pauldambra) [inline posthog/api/insight.py:378]: If I'm understanding then this only returns the first of the caching states associated with the insight being loaded. Is that necessarily in a consistent order?

I think we want to return the first of the caching states associated with this cacheable that have the cache key for the context the insight is being loaded in? 

We set that cache key in get_result as self.context.update({"filters_hash": cache_key}) so don't need to recalculate it here. (although iirc it's cheap to calculate so 🤷)

File: posthog/api/dashboard.py [hunk 2]
```diff
@@ -72,6 +72,10 @@ class Meta:
     def get_last_refresh(self, dashboard_tile: DashboardTile) -> datetime.datetime:
         if should_refresh(self.context["request"]):
             return now()
+
+        # :TODO: Remove once InsightCachingState is all populated
+        if dashboard_tile.caching_state and dashboard_tile.caching_state.last_refresh:
```
Comment (pauldambra) [inline posthog/api/dashboard.py:77]: same comment as below

**Review by pauldambra — CHANGES_REQUESTED** [review]
I either don't understand (very possible) or we need to read `caching_state` differently :)

File: posthog/api/insight.py (hunk 1 above)
Comment (macobo) [reply posthog/api/insight.py:378]: So there are 3 categories of things:
1. insight that isn't on a dashboard
2. insight that's on a dashboard
3. dashboard tiles

2 and 3 are the same.

1 and 3 have exactly 1 caching state each.

This if-else handles 3 if 3 else 1. 

You are probably reviewing this out-of-order: see https://github.com/PostHog/posthog/pull/13127 first.

Comment (pauldambra) [reply posthog/api/insight.py:378]: Ah, when saving a `DashboardTile` we trigger `update_filters_hashes` which does a [Django `bulk_update`](https://docs.djangoproject.com/en/3.2/ref/models/querysets/#bulk-update) which doesn't trigger the `post_save` signal and so the `InsightCachingState` doesn't update

You can test this by 

* creating a dashboard
* adding an insight to it
* see that there are two `InsightCachingState` rows with matching cache key
* change the dashboard date filter
* see that the `filters_hash` cache key on the  `DashboardTile` has changed but the `InsightCachingState` for that tile still has the original cache key

Comment (macobo) [reply posthog/api/insight.py:378]: See the other PR - we do listen to the same triggers when DashboardTiles are created/updated.

Comment (macobo) [reply posthog/api/insight.py:378]: Bug: Change dashboard filters, dashboard tile cached state does not get updated.

**Comment by macobo** [comment]
Superceded by https://github.com/PostHog/posthog/pull/13186

**Pull request closed without merging**